sources/platform/actors/development/actor_definition/dynamic_actor_memory/index.md

```diff
@@ -0,0 +1,193 @@
+---
+title: Dynamic Actor memory
+description: Learn how to automatically adjust your Actor's memory based on input size and run options, so you can optimize performance and reduce costs without manual configuration..
+slug: /actors/development/actor-definition/dynamic-actor-memory
+---
+
+**Learn how to automatically adjust your Actor's memory based on input size and run options, so you can optimize performance and reduce costs without manual configuration.**
+
+---
+
+Dynamic Actor memory allows Actors to _automatically adjust its memory allocation based on the input and run options_. Instead of always using a fixed memory value, Actor can use just the right amount of memory for each run.
+
+This helps:
+
+- _Optimize performance_ for large inputs (more memory for bigger tasks).
+- _Reduce costs_ for small runs (less memory when it’s not needed).
+- _Provide better user experience_, so users get optimal performance without having to manually configure memory.
+
+:::info
+_Important_: The memory calculated by dynamic expression is used as the default memory for the run. Users can still override it manually for each run.
+:::
+
+---
+
+## Why dynamic memory matters
+
+Currently, Actors often use a _static default memory_, but the optimal memory usually depends on the input:
+
+- A small input (e.g., 10 URLs) might run fine on 512 MB.
+- A large input (e.g., 1,000 URLs) could require 4 GB+ to run efficiently.
+
+Setting a single default either _wastes resources or slows down execution_. Dynamic memory solves this by calculating memory just before the run starts.
+
+---
+
+## How to define dynamic memory expression
+
+You can define a dynamic memory expression in your `actor.json`:
+
+```json
+{
+  "defaultMemoryMbytes": "get(input, 'startUrls.length' * 1024)"
+}
+```
+
+- The expression is evaluated _before the run starts_.
+- The expression can reference variables from input and run options to calculate memory (e.g., `input.startUrls`, `runOptions.maxItems`).
+- The result becomes the default memory for the run, but users can still override it.
+
+---
+
+### Writing Expressions
```

Comment:
(nit) I don't think you need this headline, it can be merged with the previous section.